linuxPackages.ply: 1.0.beta1-9e810b1 -> 2.1.1

jbcrail opened the pull request:
###### Motivation for this change

This PR upgrades the `linuxPackages.ply` package mentioned in #134265.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

jbcrail committed:
linuxPackages.ply: 1.0.beta1-9e810b1 -> 2.1.1

SuperSandro2000 commented on pkgs/os-specific/linux/ply/default.nix:
```diff
@@ -3,7 +3,7 @@
 assert kernel != null -> lib.versionAtLeast kernel.version "4.0";
 
```
```suggestion
```

Mic92 replied:
The assertion condition should be assigned to `meta.broken` so.

Mic92 commented on pkgs/os-specific/linux/ply/default.nix:
```diff
@@ -12,8 +12,8 @@ in stdenv.mkDerivation {
     repo = "ply";
-    rev = "9e810b157ba079c32c430a7d4c6034826982056e";
-    sha256 = "15cp6iczawaqlhsa0af6i37zn5iq53kh6ya8s2hzd018yd7mhg50";
+    rev = "899afb0c35ba2191dd7aa21f13bc7fde2655c475";
```
I think we can now switch to tags:

```suggestion
    rev = version;
```

jbcrail committed:
Assign assertion to meta.broken

jbcrail committed:
Switch to version tag for package revision

Mic92 commented on pkgs/os-specific/linux/ply/default.nix:
```diff
@@ -34,9 +32,10 @@ in stdenv.mkDerivation {
     homepage = "https://wkz.github.io/ply/";
     license = [ licenses.gpl2 ];
     maintainers = with maintainers; [ mic92 mbbx6spp ];
+    broken = lib.versionAtLeast kernel.version "4.0";
```
Should it not be the other way around?

```suggestion
    broken = lib.versionOlder kernel.version "4.0";
```

r-rmcgibbo commented:
Result of `nixpkgs-review pr 134266` at bcb2def4 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/c7bcd2f8e3591dbf2896d26e8f9a2f32">linuxPackages-libre.ply</a></li>
    <li><a href="https://gist.github.com/6724ac745f9e07d4ad2ec1a41f934d56">linuxPackages.ply</a> (linuxPackages_5_10.ply ,linuxPackages_5_10_hardened.ply ,linuxPackages_hardened.ply)</li>
    <li><a href="https://gist.github.com/9a4ece534a0136fbdbcad87524b19727">linuxPackages_4_14.ply</a> (linuxPackages_4_14_hardened.ply)</li>
    <li><a href="https://gist.github.com/59282546a5c3005344d31d7c2281cbbf">linuxPackages_4_19.ply</a> (linuxPackages_4_19_hardened.ply)</li>
    <li><a href="https://gist.github.com/d0353ab2278c3809f5da74e18baf3a03">linuxPackages_4_4.ply</a></li>
    <li><a href="https://gist.github.com/ddc44bfa7d756bd2452854f354685f49">linuxPackages_4_9.ply</a></li>
    <li><a href="https://gist.github.com/9a76be68280ee5565ce77616998bc9ec">linuxPackages_5_13.ply</a> (linuxPackages_latest.ply)</li>
    <li><a href="https://gist.github.com/91d10199dae0c55312b977cdc6b60a30">linuxPackages_5_4.ply</a> (linuxPackages_5_4_hardened.ply)</li>
    <li><a href="https://gist.github.com/4b64c08cfd263650c65bff4f89213d1e">linuxPackages_hardkernel_4_14.ply</a> (linuxPackages_hardkernel_latest.ply)</li>
    <li><a href="https://gist.github.com/7327a5435ac9d7924443084f875d0504">linuxPackages_latest-libre.ply</a></li>
    <li><a href="https://gist.github.com/8f5520a870a7bec09aade60fa2d55ad5">linuxPackages_lqx.ply</a></li>
    <li><a href="https://gist.github.com/5d3db752a3a797369506947ff9302b6e">linuxPackages_testing_bcachefs.ply</a></li>
    <li><a href="https://gist.github.com/e383e4d6d46e272c3e80d5ef9501b8f6">linuxPackages_xanmod.ply</a></li>
    <li><a href="https://gist.github.com/dbc9639c462b96c6a4dca128de71ae4e">linuxPackages_zen.ply</a></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/os-specific/linux/ply/default.nix:37:5:
```
   |
37 |     license = [ licenses.gpl2 ];
   |     ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

jbcrail committed:
Reverse conditional in meta.broken

jbcrail committed:
Clarify specific GPL license

jbcrail committed:
Update package version to fix build

The build for the 2.1.1 release is broken, so I updated the packge
version to use the newer commits to fix the build.

Mic92 commented:
Result of `nixpkgs-review pr 134266` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>linuxPackages-libre.ply</li>
    <li>linuxPackages.ply (linuxPackages_5_10.ply ,linuxPackages_5_10_hardened.ply ,linuxPackages_hardened.ply)</li>
    <li>linuxPackages_4_14.ply (linuxPackages_4_14_hardened.ply)</li>
    <li>linuxPackages_4_19.ply (linuxPackages_4_19_hardened.ply)</li>
    <li>linuxPackages_4_4.ply</li>
    <li>linuxPackages_4_9.ply</li>
    <li>linuxPackages_5_13.ply (linuxPackages_latest.ply)</li>
    <li>linuxPackages_5_4.ply (linuxPackages_5_4_hardened.ply)</li>
    <li>linuxPackages_hardkernel_4_14.ply (linuxPackages_hardkernel_latest.ply)</li>
    <li>linuxPackages_latest-libre.ply</li>
    <li>linuxPackages_lqx.ply</li>
    <li>linuxPackages_testing_bcachefs.ply</li>
    <li>linuxPackages_xanmod.ply</li>
    <li>linuxPackages_zen.ply</li>
  </ul>
</details>
